src/main/java/com/minecolonies/core/generation/defaults/DefaultRecipeProvider.java

```diff
@@ -439,6 +439,16 @@ private void buildFood(@NotNull final Consumer<FinishedRecipe> consumer)
                 .unlockedBy("has_dough", has(ModItems.rawPumpkinPie))
                 .save(consumer, append(new ResourceLocation(MOD_ID, "baked_pumpkin_pie"), "_", ""));
 
+        SimpleCookingRecipeBuilder.smelting(Ingredient.of(ModItems.manchet_dough),  RecipeCategory.FOOD,
+            ModItems.manchet_bread, 0.35f, 200)

```

Comment:
Atm minecolonies workers don't use smokers at all, but can do

Reply:
The convention is to have campfire = 600, smelting = 200, smoking/blasting = 100, and it looks like these already did that?

FWIW I did have a brief experiment with letting the chef using smoking recipes (if unique from smelting, which doesn't happen in vanilla but can happen with datapacks/mods); but the way their AI is constructed it wouldn't work without requiring smokers/blasters in the buildings... or changing the AI to be better at faking it.

Reply:
> The convention is to have campfire = 600, smelting = 200, smoking/blasting = 100, and it looks like these already did that?
> 
> FWIW I did have a brief experiment with letting the chef use smoking recipes (if unique from smelting, which doesn't happen in vanilla but can happen with datapacks/mods); but the way their AI is constructed it wouldn't work without requiring smokers/blasters in the buildings... or changing the AI to be better at faking it.

I wonder if we could abstract this to some "common generic type"

Reply:
In my test I was mostly just hijacking the `FurnaceRecipes` so they couldn't tell the difference, which meant they'd stuff the ingredients into the furnace and then wait forever because it would never actually start cooking.

If we did want to make the buildings require the extra blocks then it could have separate lists for smoker recipes etc, but I didn't want to pursue that since it seemed like a design pain.  Faking it (making them pull out the right item eventually even if the furnace didn't actually craft it) was another option, but that seemed messy too.

Reply:
I mean, internally these blocks work the same. And our AIs don't care about cooking time (they just idle wait and check pretty much). So we could just a) In our detection add smokers and whatever to the furnace list and b) Handle the tileentity generically in the AI and have a util method that then does smoker/furnace/etc specific handling.

Reply:
I know, the technical side isn't hard.  It's adding smokers and blasters to all the style packs that's a pain 😆

Reply:
No, what I mean is that we don't HAVE TO add them. We can add them, and we can maybe at higher building levels replace some furnaces with smokers. If the code is generic its fine with "whatever we got 5 of"